### PR TITLE
Add compatibility with vSphere

### PR DESCRIPTION
**Pull request by alexw19**
### What is this change about?

_Describe the change and why it's needed._

This change adds compatibility with vSphere

### Please provide contextual information.

_Include any links to other PRs, stories, slack discussions, etc... that will help establish context._

### Please check all that apply for this PR:
- [ ] introduces a new task
- [X] changes an existing task
- [ ] changes the Dockerfile
- [ ] introduces a breaking change (other users will need to make manual changes when this is released)

### Did you update the README as appropriate for this change?
- [ ] YES
- [X] N/A

### How should this change be described in release notes?

_Something brief that conveys the change and is written with the component author audience in mind._

Add vSphere compatibility

### What is the level of urgency for publishing this change?

- [ ] **Urgent** - unblocks current or future work
- [X] **Slightly Less than Urgent**

### Tag your pair, your PM, and/or team!
_It's helpful to tag a few other folks on your team or your team alias in case we need to follow up later._

### CODE REVIEW

File: bbl-up/task [hunk 1]
```diff
@@ -103,7 +103,7 @@ function main() {
       cp -r ${root_dir}/bbl-config/${BBL_CONFIG_DIR}/. .
     fi
 
-    eval bbl --debug up \
+    eval bbl up --debug \
```
Comment (davewalter) [inline bbl-up/task]: Could you please revert this change? The `--debug` flag is a global option according to the bbl help doc.

File: shared-functions [hunk 2]
```diff
@@ -3,7 +3,7 @@
 check_bbl_input_params() {
   set +x
   # Checks that we have IaaS credentials for GCP or AWS or Azure
-  if [ ! -z "${BBL_GCP_SERVICE_ACCOUNT_KEY}" ] || [ ! -z "${BBL_AWS_ACCESS_KEY_ID}" -a ! -z "${BBL_AWS_SECRET_ACCESS_KEY}" ] || [ ! -z "${BBL_AZURE_CLIENT_ID}" -a ! -z "${BBL_AZURE_CLIENT_SECRET}" -a ! -z "${BBL_AZURE_TENANT_ID}" -a ! -z "${BBL_AZURE_SUBSCRIPTION_ID}" ]; then
+  if [ ! -z "${BBL_GCP_SERVICE_ACCOUNT_KEY}" ] || [ ! -z "${BBL_VSPHERE_VCENTER_IP}" ] || [ ! -z "${BBL_AWS_ACCESS_KEY_ID}" -a ! -z "${BBL_AWS_SECRET_ACCESS_KEY}" ] || [ ! -z "${BBL_AZURE_CLIENT_ID}" -a ! -z "${BBL_AZURE_CLIENT_SECRET}" -a ! -z "${BBL_AZURE_TENANT_ID}" -a ! -z "${BBL_AZURE_SUBSCRIPTION_ID}" ]; then
```
Comment (davewalter) [inline shared-functions]: Is `BBL_VSPHERE_VCENTER_IP` the only required variable for vSphere? I would expect at least `BBL_VSPHERE_VCENTER_USER` and `BBL_VSPHERE_VCENTER_PASSWORD` to be required.

**Comment by Syerram** [comment]
@alexw19 Thanks for submitting the PR. Please review the requested changes by Dave. 

If we don't hear from you in the next 10 days, we'll close this PR. You can always re-open the PR.

**Comment by davewalter** [comment]
Closing due to inactivity. Please feel free to re-open if desired.
